profiler/src/ProfilerEngine/Datadog.Profiler.Native.Windows/CrashReportingWindows.cpp

```diff
@@ -0,0 +1,299 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+
+#include "windows.h"
+
+#include "CrashReportingWindows.h"
+#include "TlHelp32.h"
+#include "DbgHelp.h"
+#include "Psapi.h"
+
+#include <shared/src/native-src/string.h>
+#include <filesystem>
+
+#pragma comment(lib, "dbghelp.lib")
+
+CrashReporting* CrashReporting::Create(int32_t pid)
+{
+    auto crashReporting = new CrashReportingWindows(pid);
+    return (CrashReporting*)crashReporting;
+}
+
+CrashReportingWindows::CrashReportingWindows(int32_t pid)
+    : CrashReporting(pid)
+    , _process(NULL)
+{
+}
+
+CrashReportingWindows::~CrashReportingWindows()
+{
+    CloseHandle(_process);
+}
+
+int32_t CrashReportingWindows::Initialize()
+{
+    auto result = CrashReporting::Initialize();
+
+    if (result == 0)
+    {
+        _process = OpenProcess(PROCESS_QUERY_INFORMATION | PROCESS_VM_READ, FALSE, _pid);
+
+        if (_process == NULL)
+        {
+            return 1;
+        }
+
+        SymInitialize(_process, nullptr, TRUE);
+
+        _modules = GetModules();
+    }
+
+    return result;
+}
+
+std::vector<std::pair<int32_t, std::string>> CrashReportingWindows::GetThreads()
+{
+    std::vector<std::pair<int32_t, std::string>> threads;
+
+    auto threadSnapshot = CreateToolhelp32Snapshot(TH32CS_SNAPTHREAD, _pid);
+
+    if (threadSnapshot == INVALID_HANDLE_VALUE)
+    {
+        return threads;
+    }
+
+    THREADENTRY32 threadEntry = {};
+    threadEntry.dwSize = sizeof(THREADENTRY32);
+
+    if (Thread32First(threadSnapshot, &threadEntry))
+    {
+        do
+        {
+            if (threadEntry.th32OwnerProcessID == _pid)
+            {
+                auto thread = OpenThread(THREAD_QUERY_INFORMATION, FALSE, threadEntry.th32ThreadID);
+
+                if (thread)
+                {
+                    std::string threadName;
+                    PWSTR description;
+                    
+                    if (SUCCEEDED(GetThreadDescription(thread, &description)))
+                    {
+                        threadName = shared::ToString(description);
+                    }
+
+                    threads.push_back({ threadEntry.th32ThreadID, threadName });
+
+                    CloseHandle(thread);
+                }                
+            }
+        } while (Thread32Next(threadSnapshot, &threadEntry));
+    }
+
+    CloseHandle(threadSnapshot);
```

Comment:
Is there an issue here (or any chance of it happening) if `Thread32First` returns `false`? I assume this is correctly outside the `if` so that `CloseHandle()` is called even in that scenario, just want to check

Reply:
I don't see any issue here. We want to close the handle when we're done, whether we found threads or not.

Reply:
You're right, I misread this entirely